<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 Benchmarking analyses | An optimized practice for upstream analysis of single nucleus transcriptomics in key metabolic tissues</title>
  <meta name="description" content="7 Benchmarking analyses | An optimized practice for upstream analysis of single nucleus transcriptomics in key metabolic tissues" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="7 Benchmarking analyses | An optimized practice for upstream analysis of single nucleus transcriptomics in key metabolic tissues" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 Benchmarking analyses | An optimized practice for upstream analysis of single nucleus transcriptomics in key metabolic tissues" />
  
  
  

<meta name="author" content="Pengwei Dong" />


<meta name="date" content="2024-10-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-integration.html"/>
<link rel="next" href="reference.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">snRNAseq-analysis-workflow</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html"><i class="fa fa-check"></i><b>2</b> Genome alignment and quantification</a>
<ul>
<li class="chapter" data-level="2.1" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#installation-and-reference-downloading"><i class="fa fa-check"></i><b>2.1</b> Installation and reference downloading</a></li>
<li class="chapter" data-level="2.2" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#usage"><i class="fa fa-check"></i><b>2.2</b> Usage</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#getting-started"><i class="fa fa-check"></i><b>2.2.1</b> Getting started</a></li>
<li class="chapter" data-level="2.2.2" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#run-cellranger-count"><i class="fa fa-check"></i><b>2.2.2</b> Run cellranger count</a></li>
<li class="chapter" data-level="2.2.3" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#recommendation-for-setting-arguments"><i class="fa fa-check"></i><b>2.2.3</b> Recommendation for setting arguments</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#more-information"><i class="fa fa-check"></i><b>2.3</b> More information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html"><i class="fa fa-check"></i><b>3</b> Ambient RNA removal</a>
<ul>
<li class="chapter" data-level="3.1" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#usage-1"><i class="fa fa-check"></i><b>3.1</b> Usage</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#installation"><i class="fa fa-check"></i><b>3.1.1</b> Installation</a></li>
<li class="chapter" data-level="3.1.2" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#getting-started-1"><i class="fa fa-check"></i><b>3.1.2</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#caveats-and-hints"><i class="fa fa-check"></i><b>3.2</b> Caveats and hints</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#main-output-files"><i class="fa fa-check"></i><b>3.2.1</b> Main output files</a></li>
<li class="chapter" data-level="3.2.2" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#recommendation-for-setting-arguments-1"><i class="fa fa-check"></i><b>3.2.2</b> Recommendation for setting arguments</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#more-information-1"><i class="fa fa-check"></i><b>3.3</b> More information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="quality-control-and-doublet-removal.html"><a href="quality-control-and-doublet-removal.html"><i class="fa fa-check"></i><b>4</b> Quality control and doublet removal</a>
<ul>
<li class="chapter" data-level="4.1" data-path="quality-control-and-doublet-removal.html"><a href="quality-control-and-doublet-removal.html#library-necessary-packages"><i class="fa fa-check"></i><b>4.1</b> Library necessary packages</a></li>
<li class="chapter" data-level="4.2" data-path="quality-control-and-doublet-removal.html"><a href="quality-control-and-doublet-removal.html#setup-the-seurat-object"><i class="fa fa-check"></i><b>4.2</b> Setup the Seurat object</a></li>
<li class="chapter" data-level="4.3" data-path="quality-control-and-doublet-removal.html"><a href="quality-control-and-doublet-removal.html#dection-and-handling-of-doubletsmultiplets"><i class="fa fa-check"></i><b>4.3</b> Dection and handling of doublets/multiplets</a></li>
<li class="chapter" data-level="4.4" data-path="quality-control-and-doublet-removal.html"><a href="quality-control-and-doublet-removal.html#quality-control-and-selecting-cells-for-further-analysis"><i class="fa fa-check"></i><b>4.4</b> Quality control and selecting cells for further analysis</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="normalization-clustering-and-annotation.html"><a href="normalization-clustering-and-annotation.html"><i class="fa fa-check"></i><b>5</b> Normalization, clustering and annotation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="normalization-clustering-and-annotation.html"><a href="normalization-clustering-and-annotation.html#standard-pre-processing-workflow-by-applying-sctransform-normalization"><i class="fa fa-check"></i><b>5.1</b> Standard pre-processing workflow by applying sctransform normalization</a></li>
<li class="chapter" data-level="5.2" data-path="normalization-clustering-and-annotation.html"><a href="normalization-clustering-and-annotation.html#assigning-cell-type-identity-to-clusters"><i class="fa fa-check"></i><b>5.2</b> Assigning cell type identity to clusters</a></li>
<li class="chapter" data-level="5.3" data-path="normalization-clustering-and-annotation.html"><a href="normalization-clustering-and-annotation.html#save-the-object"><i class="fa fa-check"></i><b>5.3</b> Save the object</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>6</b> Data integration</a>
<ul>
<li class="chapter" data-level="6.1" data-path="data-integration.html"><a href="data-integration.html#library-all-the-packages"><i class="fa fa-check"></i><b>6.1</b> Library all the packages</a></li>
<li class="chapter" data-level="6.2" data-path="data-integration.html"><a href="data-integration.html#load-all-the-objects-used-for-batch-effect-correction"><i class="fa fa-check"></i><b>6.2</b> Load all the objects used for batch-effect correction</a></li>
<li class="chapter" data-level="6.3" data-path="data-integration.html"><a href="data-integration.html#pre-processing-before-data-integration"><i class="fa fa-check"></i><b>6.3</b> Pre-processing before data integration</a></li>
<li class="chapter" data-level="6.4" data-path="data-integration.html"><a href="data-integration.html#peform-analysis-without-integration"><i class="fa fa-check"></i><b>6.4</b> Peform analysis without integration</a></li>
<li class="chapter" data-level="6.5" data-path="data-integration.html"><a href="data-integration.html#perform-integration"><i class="fa fa-check"></i><b>6.5</b> Perform integration</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="data-integration.html"><a href="data-integration.html#anchor-based-cca-integration"><i class="fa fa-check"></i><b>6.5.1</b> Anchor-based CCA integration</a></li>
<li class="chapter" data-level="6.5.2" data-path="data-integration.html"><a href="data-integration.html#anchor-based-rpca-integration"><i class="fa fa-check"></i><b>6.5.2</b> Anchor-based RPCA integration</a></li>
<li class="chapter" data-level="6.5.3" data-path="data-integration.html"><a href="data-integration.html#harmony"><i class="fa fa-check"></i><b>6.5.3</b> Harmony</a></li>
<li class="chapter" data-level="6.5.4" data-path="data-integration.html"><a href="data-integration.html#scvi"><i class="fa fa-check"></i><b>6.5.4</b> scVI</a></li>
<li class="chapter" data-level="6.5.5" data-path="data-integration.html"><a href="data-integration.html#scanvi"><i class="fa fa-check"></i><b>6.5.5</b> scANVI</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="data-integration.html"><a href="data-integration.html#visualition"><i class="fa fa-check"></i><b>6.6</b> Visualition</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html"><i class="fa fa-check"></i><b>7</b> Benchmarking analyses</a>
<ul>
<li class="chapter" data-level="7.1" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#installation-1"><i class="fa fa-check"></i><b>7.1</b> Installation</a></li>
<li class="chapter" data-level="7.2" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#usage-2"><i class="fa fa-check"></i><b>7.2</b> Usage</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#load-and-preprocess-data"><i class="fa fa-check"></i><b>7.2.1</b> Load and preprocess data</a></li>
<li class="chapter" data-level="7.2.2" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#perform-the-benchmark"><i class="fa fa-check"></i><b>7.2.2</b> Perform the benchmark</a></li>
<li class="chapter" data-level="7.2.3" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#visualize-the-results"><i class="fa fa-check"></i><b>7.2.3</b> Visualize the results</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#more-information-2"><i class="fa fa-check"></i><b>7.3</b> More information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="reference.html"><a href="reference.html"><i class="fa fa-check"></i><b>8</b> Reference</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">An optimized practice for upstream analysis of single nucleus transcriptomics in key metabolic tissues</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="benchmarking-analyses" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">7</span> Benchmarking analyses<a href="benchmarking-analyses.html#benchmarking-analyses" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In addition to identification of prominent marker genes in the integrated data, benchmarking included calculations of ARI coefficients, LISI scores kBET rates for integration across different donors by using benchmarking pipeline from scib-metrices (version 0.5.1) package. All the datasets were clustered well using the above method with refined resolution.</p>
<p>We used the python package called scib that uses scanpy to streamline the integration of single-cell datasets and evaluate the results. The package contains several modules for preprocessing an anndata object, running integration methods and evaluating the resulting using a number of metrics. For preprocessing, scib.preprocessing (or scib.pp) contains functions for normalising, scaling or batch-aware selection of highly variable genes. Functions for the integration methods are in scib.integration or for short scib.ig and metrics are under scib.metrics (or scib.me).</p>
<div id="installation-1" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Installation<a href="benchmarking-analyses.html#installation-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Install the latest release on PyPI. If you want to get more information, please refer to <a href="https://scib-metrics.readthedocs.io/en/stable/index.html" class="uri">https://scib-metrics.readthedocs.io/en/stable/index.html</a></p>
<pre class="shell"><code>pip install scib-metrics</code></pre>
</div>
<div id="usage-2" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Usage<a href="benchmarking-analyses.html#usage-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="load-and-preprocess-data" class="section level3 hasAnchor" number="7.2.1">
<h3><span class="header-section-number">7.2.1</span> Load and preprocess data<a href="benchmarking-analyses.html#load-and-preprocess-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre class="shell"><code>import numpy as np
import scanpy as sc
import scvi

from scib_metrics.benchmark import Benchmarker
%matplotlib inline</code></pre>
<pre class="shell"><code>adata=sc.read(&#39;./data/DemoIntegrated/AT_integrated.h5ad&#39;)
adata</code></pre>
</div>
<div id="perform-the-benchmark" class="section level3 hasAnchor" number="7.2.2">
<h3><span class="header-section-number">7.2.2</span> Perform the benchmark<a href="benchmarking-analyses.html#perform-the-benchmark" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre class="shell"><code>bm = Benchmarker(
    adata,
    batch_key=&quot;sampleid&quot;,
    label_key=&quot;major_celltype&quot;,
    embedding_obsm_keys=[&quot;X_pca&quot;,&quot;X_integrated.dr&quot;,&quot;X_integrated.rpca&quot;,&quot;X_integrated.harmony&quot;, &quot;X_scvi&quot;, &quot;X_scanvi&quot;],
    n_jobs=6,
)
bm.benchmark()</code></pre>
</div>
<div id="visualize-the-results" class="section level3 hasAnchor" number="7.2.3">
<h3><span class="header-section-number">7.2.3</span> Visualize the results<a href="benchmarking-analyses.html#visualize-the-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre class="shell"><code>bm.plot_results_table(show=False)</code></pre>
<p><img src="image/bm.png" width="1625" /></p>
</div>
</div>
<div id="more-information-2" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> More information<a href="benchmarking-analyses.html#more-information-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If you want to get more information, please refer to the documentation and GitHub:
<a href="https://scib-metrics.readthedocs.io/en/stable/index.html" class="uri">https://scib-metrics.readthedocs.io/en/stable/index.html</a>
<a href="https://github.com/theislab/scib" class="uri">https://github.com/theislab/scib</a></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-integration.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="reference.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/00dong/snRNAseq-analysis-workflow/edit/main/07-Benchmarking.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/00dong/snRNAseq-analysis-workflow/blob/main/07-Benchmarking.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
