<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Data integration | An optimized practice for upstream analysis of single nucleus transcriptomics in key metabolic tissues</title>
  <meta name="description" content="6 Data integration | An optimized practice for upstream analysis of single nucleus transcriptomics in key metabolic tissues" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Data integration | An optimized practice for upstream analysis of single nucleus transcriptomics in key metabolic tissues" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Data integration | An optimized practice for upstream analysis of single nucleus transcriptomics in key metabolic tissues" />
  
  
  

<meta name="author" content="Pengwei Dong" />


<meta name="date" content="2024-11-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="normalization-clustering-and-annotation.html"/>
<link rel="next" href="benchmarking-analyses.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">snRNAseq-analysis-workflow</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html"><i class="fa fa-check"></i><b>2</b> Genome alignment and quantification</a>
<ul>
<li class="chapter" data-level="2.1" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#installation-and-reference-downloading"><i class="fa fa-check"></i><b>2.1</b> Installation and reference downloading</a></li>
<li class="chapter" data-level="2.2" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#usage"><i class="fa fa-check"></i><b>2.2</b> Usage</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#getting-started"><i class="fa fa-check"></i><b>2.2.1</b> Getting started</a></li>
<li class="chapter" data-level="2.2.2" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#run-cellranger-count"><i class="fa fa-check"></i><b>2.2.2</b> Run cellranger count</a></li>
<li class="chapter" data-level="2.2.3" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#recommendation-for-setting-arguments"><i class="fa fa-check"></i><b>2.2.3</b> Recommendation for setting arguments</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#more-information"><i class="fa fa-check"></i><b>2.3</b> More information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html"><i class="fa fa-check"></i><b>3</b> Ambient RNA removal</a>
<ul>
<li class="chapter" data-level="3.1" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#usage-1"><i class="fa fa-check"></i><b>3.1</b> Usage</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#installation"><i class="fa fa-check"></i><b>3.1.1</b> Installation</a></li>
<li class="chapter" data-level="3.1.2" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#getting-started-1"><i class="fa fa-check"></i><b>3.1.2</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#caveats-and-hints"><i class="fa fa-check"></i><b>3.2</b> Caveats and hints</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#main-output-files"><i class="fa fa-check"></i><b>3.2.1</b> Main output files</a></li>
<li class="chapter" data-level="3.2.2" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#recommendation-for-setting-arguments-1"><i class="fa fa-check"></i><b>3.2.2</b> Recommendation for setting arguments</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#more-information-1"><i class="fa fa-check"></i><b>3.3</b> More information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="quality-control-and-doublet-removal.html"><a href="quality-control-and-doublet-removal.html"><i class="fa fa-check"></i><b>4</b> Quality control and doublet removal</a>
<ul>
<li class="chapter" data-level="4.1" data-path="quality-control-and-doublet-removal.html"><a href="quality-control-and-doublet-removal.html#library-necessary-packages"><i class="fa fa-check"></i><b>4.1</b> Library necessary packages</a></li>
<li class="chapter" data-level="4.2" data-path="quality-control-and-doublet-removal.html"><a href="quality-control-and-doublet-removal.html#setup-the-seurat-object"><i class="fa fa-check"></i><b>4.2</b> Setup the Seurat object</a></li>
<li class="chapter" data-level="4.3" data-path="quality-control-and-doublet-removal.html"><a href="quality-control-and-doublet-removal.html#dection-and-handling-of-doubletsmultiplets"><i class="fa fa-check"></i><b>4.3</b> Dection and handling of doublets/multiplets</a></li>
<li class="chapter" data-level="4.4" data-path="quality-control-and-doublet-removal.html"><a href="quality-control-and-doublet-removal.html#quality-control-and-selecting-cells-for-further-analysis"><i class="fa fa-check"></i><b>4.4</b> Quality control and selecting cells for further analysis</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="normalization-clustering-and-annotation.html"><a href="normalization-clustering-and-annotation.html"><i class="fa fa-check"></i><b>5</b> Normalization, clustering and annotation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="normalization-clustering-and-annotation.html"><a href="normalization-clustering-and-annotation.html#standard-pre-processing-workflow-by-applying-sctransform-normalization"><i class="fa fa-check"></i><b>5.1</b> Standard pre-processing workflow by applying sctransform normalization</a></li>
<li class="chapter" data-level="5.2" data-path="normalization-clustering-and-annotation.html"><a href="normalization-clustering-and-annotation.html#assigning-cell-type-identity-to-clusters"><i class="fa fa-check"></i><b>5.2</b> Assigning cell type identity to clusters</a></li>
<li class="chapter" data-level="5.3" data-path="normalization-clustering-and-annotation.html"><a href="normalization-clustering-and-annotation.html#save-the-object"><i class="fa fa-check"></i><b>5.3</b> Save the object</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>6</b> Data integration</a>
<ul>
<li class="chapter" data-level="6.1" data-path="data-integration.html"><a href="data-integration.html#library-all-the-packages"><i class="fa fa-check"></i><b>6.1</b> Library all the packages</a></li>
<li class="chapter" data-level="6.2" data-path="data-integration.html"><a href="data-integration.html#load-all-the-objects-used-for-batch-effect-correction"><i class="fa fa-check"></i><b>6.2</b> Load all the objects used for batch-effect correction</a></li>
<li class="chapter" data-level="6.3" data-path="data-integration.html"><a href="data-integration.html#pre-processing-before-data-integration"><i class="fa fa-check"></i><b>6.3</b> Pre-processing before data integration</a></li>
<li class="chapter" data-level="6.4" data-path="data-integration.html"><a href="data-integration.html#peform-analysis-without-integration"><i class="fa fa-check"></i><b>6.4</b> Peform analysis without integration</a></li>
<li class="chapter" data-level="6.5" data-path="data-integration.html"><a href="data-integration.html#perform-integration"><i class="fa fa-check"></i><b>6.5</b> Perform integration</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="data-integration.html"><a href="data-integration.html#anchor-based-cca-integration"><i class="fa fa-check"></i><b>6.5.1</b> Anchor-based CCA integration</a></li>
<li class="chapter" data-level="6.5.2" data-path="data-integration.html"><a href="data-integration.html#anchor-based-rpca-integration"><i class="fa fa-check"></i><b>6.5.2</b> Anchor-based RPCA integration</a></li>
<li class="chapter" data-level="6.5.3" data-path="data-integration.html"><a href="data-integration.html#harmony"><i class="fa fa-check"></i><b>6.5.3</b> Harmony</a></li>
<li class="chapter" data-level="6.5.4" data-path="data-integration.html"><a href="data-integration.html#scvi"><i class="fa fa-check"></i><b>6.5.4</b> scVI</a></li>
<li class="chapter" data-level="6.5.5" data-path="data-integration.html"><a href="data-integration.html#scanvi"><i class="fa fa-check"></i><b>6.5.5</b> scANVI</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="data-integration.html"><a href="data-integration.html#visualition"><i class="fa fa-check"></i><b>6.6</b> Visualition</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html"><i class="fa fa-check"></i><b>7</b> Benchmarking analyses</a>
<ul>
<li class="chapter" data-level="7.1" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#installation-1"><i class="fa fa-check"></i><b>7.1</b> Installation</a></li>
<li class="chapter" data-level="7.2" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#usage-2"><i class="fa fa-check"></i><b>7.2</b> Usage</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#load-and-preprocess-data"><i class="fa fa-check"></i><b>7.2.1</b> Load and preprocess data</a></li>
<li class="chapter" data-level="7.2.2" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#perform-the-benchmark"><i class="fa fa-check"></i><b>7.2.2</b> Perform the benchmark</a></li>
<li class="chapter" data-level="7.2.3" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#visualize-the-results"><i class="fa fa-check"></i><b>7.2.3</b> Visualize the results</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#more-information-2"><i class="fa fa-check"></i><b>7.3</b> More information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="reference.html"><a href="reference.html"><i class="fa fa-check"></i><b>8</b> Reference</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">An optimized practice for upstream analysis of single nucleus transcriptomics in key metabolic tissues</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-integration" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">6</span> Data integration<a href="data-integration.html#data-integration" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>After filtering, mitochondrial, ribosomal protein-coding and leukocyte antigen genes were removed from these 5 datasets. Batch effect were corrected by applying following integration tools: CCA and RPCA performed in the ‘IntegrateLayers’ function which is a streamlined integrative analysis from Seurat, Harmony as well as scVI(version 1.1.2). We used all genes in scVI integration, and Seurat objects were transcribed into anndata objects using scfetch (version 0.5.0) in R with reticulate (1.37.0).</p>
<p>The integration of single-cell sequencing datasets—such as those from different experimental batches, donors, or conditions—is often a critical step in scRNA-seq workflows. Integrative analysis helps align shared cell types and states across datasets, enhancing statistical power and, more importantly, enabling accurate comparative analysis. In previous versions of Seurat, we introduced anchor-based integration methods to facilitate this process. Additionally, several labs have developed innovative tools for integration, including Harmony and scVI, which have proven to be powerful methods for scRNA-seq analysis. For more information, please refer to our vignette on integrating scRNA-seq data using multiple tools.</p>
<div id="library-all-the-packages" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Library all the packages<a href="data-integration.html#library-all-the-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre class="shell"><code>library(reticulate)
library(scfetch)
library(Seurat)
library(SeuratData)
library(harmony)
library(dplyr)
library(patchwork)
library(sctransform)
library(ggplot2)</code></pre>
</div>
<div id="load-all-the-objects-used-for-batch-effect-correction" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Load all the objects used for batch-effect correction<a href="data-integration.html#load-all-the-objects-used-for-batch-effect-correction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre class="shell"><code>cohort_1 &lt;- readRDS(&#39;./data/DemoIntegrated/cohort_1_s.rds&#39;)
cohort_2_s_1 &lt;- readRDS(&#39;./data/DemoIntegrated/cohort_2_s_1_s.rds&#39;)
cohort_2_s_2 &lt;- readRDS(&#39;./data/DemoIntegrated/cohort_2_s_2_s.rds&#39;)
cohort_3 &lt;- readRDS(&#39;./data/DemoIntegrated/cohort_3_s.rds&#39;)
cohort_4_o_1 &lt;- readRDS(&#39;./data/DemoIntegrated/cohort_4_o_1_s.rds&#39;)
cohort_4_o_2 &lt;- readRDS(&#39;./data/DemoIntegrated/cohort_4_o_2_s.rds&#39;)
cohort_4_o_3 &lt;- readRDS(&#39;./data/DemoIntegrated/cohort_4_o_3_s.rds&#39;)
cohort_4_o_4 &lt;- readRDS(&#39;./data/DemoIntegrated/cohort_4_o_4_s.rds&#39;)
cohort_4_o_5 &lt;- readRDS(&#39;./data/DemoIntegrated/cohort_4_o_5_s.rds&#39;)</code></pre>
</div>
<div id="pre-processing-before-data-integration" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Pre-processing before data integration<a href="data-integration.html#pre-processing-before-data-integration" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Add Donor ID as batch information</p>
<pre class="shell"><code>cohort_1@meta.data$sampleid &lt;- &#39;ID1&#39;
cohort_3@meta.data$sampleid &lt;- &#39;ID2&#39;
cohort_2_s_1@meta.data$sampleid &lt;- &#39;ID3&#39;
cohort_2_s_2@meta.data$sampleid &lt;- &#39;ID4&#39;
cohort_4_o_1@meta.data$sampleid &lt;- &#39;ID5&#39;
cohort_4_o_2@meta.data$sampleid &lt;- &#39;ID6&#39;
cohort_4_o_3@meta.data$sampleid &lt;- &#39;ID7&#39;
cohort_4_o_4@meta.data$sampleid &lt;- &#39;ID8&#39;
cohort_4_o_5@meta.data$sampleid &lt;- &#39;ID9&#39;</code></pre>
<p>Create a object which contains multiple ‘layers’ to help to integrate all samples together, so that we can jointly identify cell subpopulations across datasets, and then explore how each group differs across conditions.</p>
<pre class="shell"><code>NC_list &lt;- list(cohort_1, cohort_2_s_1, cohort_2_s_2, cohort_3,
                cohort_4_o_1, cohort_4_o_2, cohort_4_o_3, cohort_4_o_4, cohort_4_o_5)

for (i in 1:length(NC_list)) {
  DefaultAssay(NC_list[[i]]) &lt;- &#39;RNA&#39;
  NC_list[[i]] &lt;- SCTransform(NC_list[[i]], vars.to.regress = &quot;percent.mt&quot;, verbose = FALSE)
}

NC_merged &lt;- merge(x = NC_list[[1]],
                     y = NC_list[2:length(NC_list)])</code></pre>
<p>In order to better identify the cell types, we plan to remove MT-/RP-/HLA- related genes before integration</p>
<pre class="shell"><code>gene_mt &lt;- rownames(NC_merged)[grep(&quot;^MT-&quot;, rownames(NC_merged))]
gene_rb &lt;- rownames(NC_merged)[grep(&quot;^RP[SL]&quot;, rownames(NC_merged))]
gene_hla &lt;- rownames(NC_merged)[grep(&quot;^HLA-&quot;, rownames(NC_merged))]

gene_rb &lt;- as.data.frame(gene_rb)
gene_hla &lt;- as.data.frame(gene_hla)
gene_mt &lt;- as.data.frame(gene_mt)
gene &lt;- rownames(NC_merged)
gene &lt;- as.data.frame(gene)

gene$type &lt;- gene$gene %in% gene_rb$gene_rb
gene &lt;- gene[gene$type == &quot;FALSE&quot;,]

gene$type &lt;- gene$gene %in% gene_hla$gene_hla
gene &lt;- gene[gene$type == &quot;FALSE&quot;,]

gene$type &lt;- gene$gene %in% gene_mt$gene_mt
gene &lt;- gene[gene$type == &quot;FALSE&quot;,]
gene_list &lt;- gene$gene

for (i in 1: length(NC_list)) {
  DefaultAssay(NC_list[[i]]) &lt;- &#39;RNA&#39;
  counts &lt;- GetAssayData(NC_list[[i]], layer = &#39;counts&#39;)
  counts &lt;- counts[rownames(counts) %in% gene_list, ]
  meta.data &lt;- NC_list[[i]]@meta.data
  NC_list[[i]] &lt;- CreateSeuratObject(counts = counts,
                                       meta.data = meta.data)
  
  NC_list[[i]] &lt;- SCTransform(NC_list[[i]], vars.to.regress = &quot;percent.mt&quot;, verbose = FALSE)
}

NC_merged &lt;- merge(NC_list[[1]], 
                     y = NC_list[2:length(NC_list)],
                     merge.data = TRUE)
NC_merged</code></pre>
</div>
<div id="peform-analysis-without-integration" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> Peform analysis without integration<a href="data-integration.html#peform-analysis-without-integration" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We begin by analyzing the dataset without performing integration to assess whether batch-effect correction is necessary. The resulting clusters are defined by both cell type and stimulation condition, which poses challenges for downstream analysis.</p>
<pre class="shell"><code>DefaultAssay(NC_merged) &lt;- &#39;SCT&#39;

NC_merged &lt;- SCTransform(NC_merged, verbose = T) 

NC_merged &lt;- RunPCA(NC_merged, npcs = 100, verbose = T)
ElbowPlot(NC_merged, ndims = 50)

NC_merged &lt;- RunUMAP(NC_merged, dims = 1:30, reduction = &quot;pca&quot;, reduction.name = &quot;unintegrated&quot;)

DimPlot(NC_merged, reduction = &quot;unintegrated&quot;, group.by = &quot;sampleid&quot;)</code></pre>
<p><img src="image/unintegrated.png" width="824" /></p>
</div>
<div id="perform-integration" class="section level2 hasAnchor" number="6.5">
<h2><span class="header-section-number">6.5</span> Perform integration<a href="data-integration.html#perform-integration" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We now aim to integrate data from all the donors, so that cells from the same cell type/subpopulation will cluster together.For data integration, our goal is not to remove biological differences across conditions or different batches, but to learn shared cell types/states in an initial step - specifically because that will enable us to compare control stimulated and control profiles for these individual cell types.</p>
<p>Seurat v5 supports a more streamlined integrated analysis through the use of the IntegrateLayers function. The current method supports five integration methods. Each of these methods performs integration in a low-dimensional space and returns a dimensionality reduction (i.e. integrated.rpca) designed to embed shared unit types across batches, here we use three of them:</p>
<ol style="list-style-type: decimal">
<li><p>Anchor-based CCA integration (method=CCAIntegration)</p></li>
<li><p>Anchor-based RPCA integration (method=RPCAIntegration)</p></li>
<li><p>Harmony (method=HarmonyIntegration)</p></li>
</ol>
<div id="anchor-based-cca-integration" class="section level3 hasAnchor" number="6.5.1">
<h3><span class="header-section-number">6.5.1</span> Anchor-based CCA integration<a href="data-integration.html#anchor-based-cca-integration" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The Seurat v5 integration procedure aims to return a single dimensional reduction that captures the shared sources of variance across multiple layers, so that cells in a similar biological state will cluster. The method returns a dimensional reduction (i.e. integrated.cca) which can be used for visualization and unsupervised clustering analysis. For evaluating performance, we can use cell type labels that are pre-loaded in the seurat_annotations metadata column.</p>
<pre class="shell"><code>NC_merged &lt;- IntegrateLayers(
  object = NC_merged, method = CCAIntegration,
  normalization.method = &quot;SCT&quot;, 
  verbose = FALSE)

NC_merged &lt;- FindNeighbors(NC_merged, reduction = &quot;integrated.dr&quot;, dims = 1:30, verbose = FALSE)
NC_merged &lt;- FindClusters(NC_merged, verbose = FALSE)
NC_merged &lt;- RunUMAP(NC_merged, reduction = &quot;integrated.dr&quot;, dims = 1:30, reduction.name = &quot;cca&quot;, verbose = FALSE)</code></pre>
</div>
<div id="anchor-based-rpca-integration" class="section level3 hasAnchor" number="6.5.2">
<h3><span class="header-section-number">6.5.2</span> Anchor-based RPCA integration<a href="data-integration.html#anchor-based-rpca-integration" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre class="shell"><code>NC_merged &lt;- IntegrateLayers(
  object = NC_merged, method = RPCAIntegration,
  normalization.method = &quot;SCT&quot;, 
  new.reduction = &quot;integrated.rpca&quot;,
  verbose = FALSE)

NC_merged &lt;- FindNeighbors(NC_merged, reduction = &quot;integrated.rpca&quot;, dims = 1:30, verbose = FALSE)
NC_merged &lt;- FindClusters(NC_merged, verbose = FALSE)
NC_merged &lt;- RunUMAP(NC_merged, reduction = &quot;integrated.rpca&quot;, dims = 1:30, reduction.name = &quot;rpca&quot;, verbose = FALSE)</code></pre>
</div>
<div id="harmony" class="section level3 hasAnchor" number="6.5.3">
<h3><span class="header-section-number">6.5.3</span> Harmony<a href="data-integration.html#harmony" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre class="shell"><code>NC_merged &lt;- IntegrateLayers(
  object = NC_merged, method = HarmonyIntegration,
  normalization.method = &quot;SCT&quot;, 
  new.reduction = &quot;integrated.harmony&quot;,
  verbose = FALSE)

NC_merged &lt;- FindNeighbors(NC_merged, reduction = &quot;integrated.harmony&quot;, dims = 1:30, verbose = FALSE)
NC_merged &lt;- FindClusters(NC_merged, verbose = FALSE)
NC_merged &lt;- RunUMAP(NC_merged, reduction = &quot;integrated.harmony&quot;, dims = 1:30, reduction.name = &quot;harmony&quot;, verbose = FALSE)</code></pre>
</div>
<div id="scvi" class="section level3 hasAnchor" number="6.5.4">
<h3><span class="header-section-number">6.5.4</span> scVI<a href="data-integration.html#scvi" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>scvi-tools (single-cell variational inference tools) is a package for probabilistic modeling of single-cell omics data, built on top of PyTorch and AnnData. The package hosts implementations of several models that perform a wide range of single-cell data analysis tasks, as well as the building blocks to rapidly prototype new probabilistic models.</p>
<p>For using scVI in R, scVI integration requires installing reticulate and scfetch as well as scvi-tools and its depencies in its environment. You can find the detailed instructions on the documentation:
<a href="https://scvi-tools.org/get_started/" class="uri">https://scvi-tools.org/get_started/</a></p>
<pre class="shell"><code>use_condaenv(&quot;~/miniconda3/envs/scvi&quot;, required=T)
sc &lt;- import(&quot;scanpy&quot;, convert = FALSE)
scvi &lt;- import(&quot;scvi&quot;, convert = FALSE)

# prepare an anndata
NC_merged@assays$SCT@data &lt;- NC_merged@assays$SCT@counts
dim(NC_merged@assays$SCT@data)
ExportSeurat(seu.obj = NC_merged, assay = &quot;SCT&quot;, to = &quot;AnnData&quot;,
             conda.path = &quot;~/miniconda3/envs/scvi&quot;,
             anndata.file = &quot;./data/DemoIntegrated/merged.h5ad&quot;)

adata &lt;- sc$read(&#39;./data/DemoIntegrated/merged.h5ad&#39;)

# run setup_anndata
scvi$model$SCVI$setup_anndata(adata, batch_key = &#39;sampleid&#39;) 
model = scvi$model$SCVI(adata)

# train the model
model$train()
#  get the latent represention
latent = model$get_latent_representation()

# put it back in the original Seurat object
latent &lt;- as.matrix(latent)
rownames(latent) = colnames(NC_merged)
NC_merged[[&quot;integrated.scvi&quot;]] &lt;- CreateDimReducObject(embeddings = latent, key = &quot;scvi_&quot;, assay = DefaultAssay(NC_merged))

NC_merged &lt;- FindNeighbors(NC_merged, reduction = &quot;integrated.scvi&quot;, dims = 1:10, verbose = FALSE)
NC_merged &lt;- FindClusters(NC_merged, resolution = 0.6, verbose = FALSE)

NC_merged &lt;- RunUMAP(NC_merged, dims = 1:10, reduction = &#39;integrated.scvi&#39;, reduction.name = &#39;scvi&#39;, verbose = FALSE)</code></pre>
</div>
<div id="scanvi" class="section level3 hasAnchor" number="6.5.5">
<h3><span class="header-section-number">6.5.5</span> scANVI<a href="data-integration.html#scanvi" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre class="shell"><code># run setup_anndata
model = scvi$model$SCANVI(adata)

# train the model
model$train()
#  get the latent represention
latent = model$get_latent_representation()

# put it back in the original Seurat object
latent &lt;- as.matrix(latent)
rownames(latent) = colnames(NC_merged)
NC_merged[[&quot;integrated.scanvi&quot;]] &lt;- CreateDimReducObject(embeddings = latent, key = &quot;scanvi_&quot;, assay = DefaultAssay(NC_merged))

NC_merged &lt;- FindNeighbors(NC_merged, reduction = &quot;integrated.scanvi&quot;, dims = 1:10, verbose = FALSE)
NC_merged &lt;- FindClusters(NC_merged, resolution = 0.6, verbose = FALSE)

NC_merged &lt;- RunUMAP(NC_merged, dims = 1:10, reduction = &#39;integrated.scanvi&#39;, reduction.name = &#39;scanvi&#39;, verbose = FALSE)</code></pre>
</div>
</div>
<div id="visualition" class="section level2 hasAnchor" number="6.6">
<h2><span class="header-section-number">6.6</span> Visualition<a href="data-integration.html#visualition" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre class="shell"><code>reduction_name &lt;- c(&#39;unintegrated&#39;,&#39;rpca&#39;,&#39;cca&#39;,&#39;harmony&#39;,&#39;scvi&#39;,&#39;scanvi&#39;)
for (i in 1:length(reduction_name)) {
  a &lt;- DimPlot(NC_merged, reduction = reduction_name[i], label = TRUE) + NoLegend()
  b &lt;- DimPlot(NC_merged, reduction = reduction_name[i], group.by = &quot;sampleid&quot;)
  c &lt;- DimPlot(NC_merged, reduction = reduction_name[i], group.by = &quot;celltype&quot;)
  
  print(a+b+c)
}
</code></pre>
<p><img src="image/unintegrated_a.png" width="957" /><img src="image/cca.png" width="957" /><img src="image/rpca.png" width="957" /><img src="image/harmony.png" width="957" /><img src="image/scvi.png" width="957" /><img src="image/scanvi.png" width="957" /></p>
<pre><code>## R version 4.3.2 (2023-10-31)
## Platform: x86_64-apple-darwin20 (64-bit)
## Running under: macOS Monterey 12.4
## 
## Matrix products: default
## BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
## LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## time zone: Asia/Shanghai
## tzcode source: internal
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## loaded via a namespace (and not attached):
##  [1] gtable_0.3.5      jsonlite_1.8.9    dplyr_1.1.4       compiler_4.3.2    highr_0.11        tidyselect_1.2.1  gridExtra_2.3    
##  [8] jquerylib_0.1.4   scales_1.3.0      png_0.1-8         yaml_2.3.10       fastmap_1.2.0     ggplot2_3.5.1     R6_2.5.1         
## [15] generics_0.1.3    knitr_1.48        viridis_0.6.5     tibble_3.2.1      bookdown_0.40     munsell_0.5.1     bslib_0.8.0      
## [22] pillar_1.9.0      rlang_1.1.4       utf8_1.2.4        cachem_1.1.0      xfun_0.47         sass_0.4.9        viridisLite_0.4.2
## [29] cli_3.6.3         magrittr_2.0.3    digest_0.6.37     grid_4.3.2        rstudioapi_0.16.0 lifecycle_1.0.4   vctrs_0.6.5      
## [36] evaluate_1.0.0    glue_1.7.0        rsconnect_1.3.1   fansi_1.0.6       colorspace_2.1-1  rmarkdown_2.28    tools_4.3.2      
## [43] pkgconfig_2.0.3   htmltools_0.5.8.1</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="normalization-clustering-and-annotation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="benchmarking-analyses.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/00dong/snRNAseq-analysis-workflow/edit/main/06-Integration.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/00dong/snRNAseq-analysis-workflow/blob/main/06-Integration.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
