<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Normalization, clustering and annotation | An optimized practice for upstream analysis of single nucleus transcriptomics in key metabolic tissues</title>
  <meta name="description" content="5 Normalization, clustering and annotation | An optimized practice for upstream analysis of single nucleus transcriptomics in key metabolic tissues" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Normalization, clustering and annotation | An optimized practice for upstream analysis of single nucleus transcriptomics in key metabolic tissues" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Normalization, clustering and annotation | An optimized practice for upstream analysis of single nucleus transcriptomics in key metabolic tissues" />
  
  
  

<meta name="author" content="Pengwei Dong" />


<meta name="date" content="2024-10-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="quality-control-and-doublet-removal.html"/>
<link rel="next" href="data-integration.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">snRNAseq-analysis-workflow</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html"><i class="fa fa-check"></i><b>2</b> Genome alignment and quantification</a>
<ul>
<li class="chapter" data-level="2.1" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#installation-and-reference-downloading"><i class="fa fa-check"></i><b>2.1</b> Installation and reference downloading</a></li>
<li class="chapter" data-level="2.2" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#usage"><i class="fa fa-check"></i><b>2.2</b> Usage</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#getting-started"><i class="fa fa-check"></i><b>2.2.1</b> Getting started</a></li>
<li class="chapter" data-level="2.2.2" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#run-cellranger-count"><i class="fa fa-check"></i><b>2.2.2</b> Run cellranger count</a></li>
<li class="chapter" data-level="2.2.3" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#recommendation-for-setting-arguments"><i class="fa fa-check"></i><b>2.2.3</b> Recommendation for setting arguments</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="genome-alignment-and-quantification.html"><a href="genome-alignment-and-quantification.html#more-information"><i class="fa fa-check"></i><b>2.3</b> More information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html"><i class="fa fa-check"></i><b>3</b> Ambient RNA removal</a>
<ul>
<li class="chapter" data-level="3.1" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#usage-1"><i class="fa fa-check"></i><b>3.1</b> Usage</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#installation"><i class="fa fa-check"></i><b>3.1.1</b> Installation</a></li>
<li class="chapter" data-level="3.1.2" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#getting-started-1"><i class="fa fa-check"></i><b>3.1.2</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#caveats-and-hints"><i class="fa fa-check"></i><b>3.2</b> Caveats and hints</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#main-output-files"><i class="fa fa-check"></i><b>3.2.1</b> Main output files</a></li>
<li class="chapter" data-level="3.2.2" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#recommendation-for-setting-arguments-1"><i class="fa fa-check"></i><b>3.2.2</b> Recommendation for setting arguments</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="ambient-rna-removal.html"><a href="ambient-rna-removal.html#more-information-1"><i class="fa fa-check"></i><b>3.3</b> More information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="quality-control-and-doublet-removal.html"><a href="quality-control-and-doublet-removal.html"><i class="fa fa-check"></i><b>4</b> Quality control and doublet removal</a>
<ul>
<li class="chapter" data-level="4.1" data-path="quality-control-and-doublet-removal.html"><a href="quality-control-and-doublet-removal.html#library-necessary-packages"><i class="fa fa-check"></i><b>4.1</b> Library necessary packages</a></li>
<li class="chapter" data-level="4.2" data-path="quality-control-and-doublet-removal.html"><a href="quality-control-and-doublet-removal.html#setup-the-seurat-object"><i class="fa fa-check"></i><b>4.2</b> Setup the Seurat object</a></li>
<li class="chapter" data-level="4.3" data-path="quality-control-and-doublet-removal.html"><a href="quality-control-and-doublet-removal.html#dection-and-handling-of-doubletsmultiplets"><i class="fa fa-check"></i><b>4.3</b> Dection and handling of doublets/multiplets</a></li>
<li class="chapter" data-level="4.4" data-path="quality-control-and-doublet-removal.html"><a href="quality-control-and-doublet-removal.html#quality-control-and-selecting-cells-for-further-analysis"><i class="fa fa-check"></i><b>4.4</b> Quality control and selecting cells for further analysis</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="normalization-clustering-and-annotation.html"><a href="normalization-clustering-and-annotation.html"><i class="fa fa-check"></i><b>5</b> Normalization, clustering and annotation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="normalization-clustering-and-annotation.html"><a href="normalization-clustering-and-annotation.html#standard-pre-processing-workflow-by-applying-sctransform-normalization"><i class="fa fa-check"></i><b>5.1</b> Standard pre-processing workflow by applying sctransform normalization</a></li>
<li class="chapter" data-level="5.2" data-path="normalization-clustering-and-annotation.html"><a href="normalization-clustering-and-annotation.html#assigning-cell-type-identity-to-clusters"><i class="fa fa-check"></i><b>5.2</b> Assigning cell type identity to clusters</a></li>
<li class="chapter" data-level="5.3" data-path="normalization-clustering-and-annotation.html"><a href="normalization-clustering-and-annotation.html#save-the-object"><i class="fa fa-check"></i><b>5.3</b> Save the object</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-integration.html"><a href="data-integration.html"><i class="fa fa-check"></i><b>6</b> Data integration</a>
<ul>
<li class="chapter" data-level="6.1" data-path="data-integration.html"><a href="data-integration.html#library-all-the-packages"><i class="fa fa-check"></i><b>6.1</b> Library all the packages</a></li>
<li class="chapter" data-level="6.2" data-path="data-integration.html"><a href="data-integration.html#load-all-the-objects-used-for-batch-effect-correction"><i class="fa fa-check"></i><b>6.2</b> Load all the objects used for batch-effect correction</a></li>
<li class="chapter" data-level="6.3" data-path="data-integration.html"><a href="data-integration.html#pre-processing-before-data-integration"><i class="fa fa-check"></i><b>6.3</b> Pre-processing before data integration</a></li>
<li class="chapter" data-level="6.4" data-path="data-integration.html"><a href="data-integration.html#peform-analysis-without-integration"><i class="fa fa-check"></i><b>6.4</b> Peform analysis without integration</a></li>
<li class="chapter" data-level="6.5" data-path="data-integration.html"><a href="data-integration.html#perform-integration"><i class="fa fa-check"></i><b>6.5</b> Perform integration</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="data-integration.html"><a href="data-integration.html#anchor-based-cca-integration"><i class="fa fa-check"></i><b>6.5.1</b> Anchor-based CCA integration</a></li>
<li class="chapter" data-level="6.5.2" data-path="data-integration.html"><a href="data-integration.html#anchor-based-rpca-integration"><i class="fa fa-check"></i><b>6.5.2</b> Anchor-based RPCA integration</a></li>
<li class="chapter" data-level="6.5.3" data-path="data-integration.html"><a href="data-integration.html#harmony"><i class="fa fa-check"></i><b>6.5.3</b> Harmony</a></li>
<li class="chapter" data-level="6.5.4" data-path="data-integration.html"><a href="data-integration.html#scvi"><i class="fa fa-check"></i><b>6.5.4</b> scVI</a></li>
<li class="chapter" data-level="6.5.5" data-path="data-integration.html"><a href="data-integration.html#scanvi"><i class="fa fa-check"></i><b>6.5.5</b> scANVI</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="data-integration.html"><a href="data-integration.html#visualition"><i class="fa fa-check"></i><b>6.6</b> Visualition</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html"><i class="fa fa-check"></i><b>7</b> Benchmarking analyses</a>
<ul>
<li class="chapter" data-level="7.1" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#installation-1"><i class="fa fa-check"></i><b>7.1</b> Installation</a></li>
<li class="chapter" data-level="7.2" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#usage-2"><i class="fa fa-check"></i><b>7.2</b> Usage</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#load-and-preprocess-data"><i class="fa fa-check"></i><b>7.2.1</b> Load and preprocess data</a></li>
<li class="chapter" data-level="7.2.2" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#perform-the-benchmark"><i class="fa fa-check"></i><b>7.2.2</b> Perform the benchmark</a></li>
<li class="chapter" data-level="7.2.3" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#visualize-the-results"><i class="fa fa-check"></i><b>7.2.3</b> Visualize the results</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="benchmarking-analyses.html"><a href="benchmarking-analyses.html#more-information-2"><i class="fa fa-check"></i><b>7.3</b> More information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="reference.html"><a href="reference.html"><i class="fa fa-check"></i><b>8</b> Reference</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">An optimized practice for upstream analysis of single nucleus transcriptomics in key metabolic tissues</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="normalization-clustering-and-annotation" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">5</span> Normalization, clustering and annotation<a href="normalization-clustering-and-annotation.html#normalization-clustering-and-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Gene-expression values were normalized using sctransform (v2). Principal component analysis (PCA) was carried out using the variable features for the SCT Assay Uniform Manifold Approximation and Projection (UMAP) analysis was performed to further reduce variation to two dimensions with ‘RunUMAP’ function with 25 to 40 principal components, and a resolution range from 0.6 to 2. These clusters were manually annotated based on differentially expressed genes generated from ‘FindAllMarkers’ function as well as the canonical marker genes from published literature.</p>
<div id="standard-pre-processing-workflow-by-applying-sctransform-normalization" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Standard pre-processing workflow by applying sctransform normalization<a href="normalization-clustering-and-annotation.html#standard-pre-processing-workflow-by-applying-sctransform-normalization" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The sctransform function in Seurat consolidates the functionality of NormalizeData(), ScaleData(), and FindVariableFeatures() into a single command. The transformed data is stored in the SCT assay, which is set as the default assay after running sctransform.</p>
<p>During normalization, it is possible to regress out confounding sources of variation, such as the percentage of mitochondrial reads. In Seurat v5, SCT v2 is applied by default, though you can revert to v1 by setting vst.flavor = ‘v1’. Additionally, the glmGamPoi package, which significantly enhances processing speed, is used by default if installed. You can find installation instructions here.</p>
<p>In the same time, we perform dimensional reduction and genarates an ‘Elbow plot’ to find the majority of true signal captured in each PCs</p>
<pre class="shell"><code>cohort_1 &lt;- SCTransform(cohort_1, vars.to.regress = &quot;percent.mt&quot;, verbose = FALSE) %&gt;%
   RunPCA(npcs = 75, verbose = FALSE)
ElbowPlot(cohort_1, ndims = 50)</code></pre>
<p><img src="image/elbow.png" width="762" /></p>
<p>To cluster the cells, we apply modularity optimization techniques, such as the Louvain algorithm (default), which iteratively group cells with the goal of optimizing the standard modularity function. The FindClusters() function in Seurat implements this procedure and includes a resolution parameter that controls the ‘granularity’ of clustering. Higher resolution values result in a greater number of clusters, providing finer distinctions between cell groups.</p>
<pre class="shell"><code>cohort_1 &lt;- RunUMAP(cohort_1, reduction = &quot;pca&quot;, dims = 1:40, verbose = FALSE) %&gt;%
  FindNeighbors(reduction = &quot;pca&quot;, dims = 1:40, verbose = FALSE) %&gt;%
  FindClusters(resolution = 0.6, verbose = FALSE)
DimPlot(cohort_1, label = T, repel = T) </code></pre>
<p><img src="image/umap.png" width="730" /></p>
</div>
<div id="assigning-cell-type-identity-to-clusters" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Assigning cell type identity to clusters<a href="normalization-clustering-and-annotation.html#assigning-cell-type-identity-to-clusters" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Fortunately in the case of the adipose tissue dataset, we can use canonical markers to easily match the unbiased clustering to known cell types:</p>
<pre class="shell"><code>genes_to_check &lt;- c (&quot;CD3D&quot;, &quot;CD3E&quot;, &quot;PTPRC&quot;, &quot;CD4&quot;, 
                     &quot;CD2&quot;, &quot;CD40LG&quot;, &quot;IL7R&quot;, #Naive CD4 T
                     &quot;KLRF1&quot;, &quot;PRF1&quot;, &quot;GZMB&quot;, &quot;KLRD1&quot;,#Cytotoxic CD8 T
                     &quot;CD8A&quot;, &quot;GZMH&quot;, &quot;NKG7&quot;, #Naive CD8 T
                     &quot;TIGIT&quot;, &quot;FOXP3&quot;, #Treg
                     &quot;CCR6&quot;, &quot;KLRB1&quot;, &quot;CEBPD&quot;, #MAIT
                     &quot;LEF1&quot;, &quot;KLRC2&quot;, &quot;ZNF683&quot;, #CD8 gd T
                     &quot;GZMK&quot;, &quot;CCL5&quot;, &quot;XCL2&quot;, &quot;CCL4&quot;, #NK-like
                     &quot;CX3CR1&quot;, &quot;GNLY&quot;, &quot;SPON2&quot;, #mNK
                     &quot;KIT&quot;, &quot;CD200R1&quot;,  #ILCs
                     &quot;SMIM25&quot;, #ncMo
                     &quot;S100A9&quot;, &quot;MMP19&quot;, #Monocyte
                     &quot;S100A8&quot;, &quot;EREG&quot;, &quot;IL1B&quot;, #myeloid-like
                     &quot;XCR1&quot;, &quot;CLEC9A&quot;, &quot;CADM1&quot;, #cDC1
                     &quot;CLEC10A&quot;, &quot;CD1C&quot;, &quot;FCER1A&quot;, #cDC2B
                     &quot;IGKC&quot;, &quot;CD79A&quot;, #B cell
                     &quot;C1QA&quot;, &quot;RNASE1&quot;, # PVMs
                     &quot;MRC1&quot;, &quot;C1QB&quot;, &quot;CD86&quot;, #Macrophage
                     &quot;DKK1&quot;, &quot;PI16&quot;, &quot;PRG4&quot;, &quot;DPP4&quot;,&quot;PDGFRA&quot;, #APC
                     &quot;RGS5&quot;,&quot;MYH11&quot;, #SMC &amp; Pericyte
                     &quot;CLDN5&quot;, &quot;VWF&quot;, &quot;PECAM1&quot;, #Endothelial
                     &quot;PLIN1&quot;,&quot;PLIN4&quot;,&quot;ADIPOQ&quot;) #Adipocyte

DotPlot(cohort_1, features = genes_to_check,  
        cols = c(&quot;gray95&quot;, &quot;red&quot;), cluster.idents = T, col.min = 0, 
        assay=&#39;SCT&#39;) + coord_flip()</code></pre>
<p><img src="image/dotplot.png" width="744" /></p>
<pre class="shell"><code>cohort_1@meta.data$major_celltype[cohort_1@meta.data$seurat_clusters %in% c(0,10,12)] &lt;- &#39;Adipocyte&#39;
cohort_1@meta.data$major_celltype[cohort_1@meta.data$seurat_clusters %in% c(8,9)] &lt;- &#39;SMC&#39;
cohort_1@meta.data$major_celltype[cohort_1@meta.data$seurat_clusters %in% 1] &lt;- &#39;Macrophage&#39;
cohort_1@meta.data$major_celltype[cohort_1@meta.data$seurat_clusters %in% c(2,7)] &lt;- &#39;FAP&#39;
cohort_1@meta.data$major_celltype[cohort_1@meta.data$seurat_clusters %in% 11] &lt;- &#39;T&#39;
cohort_1@meta.data$major_celltype[cohort_1@meta.data$seurat_clusters %in% c(3,4,5,13)] &lt;- &#39;EC&#39;
cohort_1@meta.data$major_celltype[cohort_1@meta.data$seurat_clusters %in% 6] &lt;- &#39;Monocyte&#39;
cohort_1@meta.data$major_celltype[cohort_1@meta.data$seurat_clusters %in% 14] &lt;- &#39;ILC&#39;
DimPlot(cohort_1, group.by = &#39;major_celltype&#39;)</code></pre>
<p><img src="image/annotated.png" width="546" /></p>
</div>
<div id="save-the-object" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Save the object<a href="normalization-clustering-and-annotation.html#save-the-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre class="shell"><code>saveRDS(cohort_1, file = &#39;./data/DemoSingle/Cohort_1_demo_annotated.rds&#39;)</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="quality-control-and-doublet-removal.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="data-integration.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/00dong/snRNAseq-analysis-workflow/edit/main/05-Normalization.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/00dong/snRNAseq-analysis-workflow/blob/main/05-Normalization.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
